src/imagery/i.eodag/i.eodag.py

```diff
@@ -0,0 +1,326 @@
+                "lonmax": 2,
+                "latmax": 44,
+            }  # hardcoded for testing
+        )
+
+        gs.verbose(geom)
+
+        search_parameters = {
+            "items_per_page": items_per_page,
+            "productType": product_type,
+            # TODO: Convert to a shapely object
+            "geom": geom,
+        }
+
+        if options["clouds"]:
+            search_parameters["cloudCover"] = options["clouds"]
+
+        start_date = options["start"]
+        if not options["start"]:
+            delta_days = timedelta(60)
+            start_date = str(datetime.now() - delta_days)
+        start_date = datetime.fromisoformat(start_date)
+
+        end_date = options["end"]
+        if not options["end"]:
+            end_date = str(datetime.now())
+        end_date = datetime.fromisoformat(end_date)
+
+        if end_date < start_date:
+            gs.fatal(
+                _(f"End Date ({end_date}) cannot come before Start Date ({start_date})")
+            )
+
+        # Requires further testing to make sure the isoformat works with all the providers
+        # TODO: Normalize to UTC
+        search_parameters["start"] = start_date.isoformat()[:-6]  # Remove the timezone
+        search_parameters["end"] = end_date.isoformat()[:-6]
+
+        search_results = dag.search_all(**search_parameters)
+        num_results = len(search_results)
+        gs.message(_(f"Found {num_results} matching scenes " f"of type {product_type}"))
+        if flags["l"]:
+            # TODO: Oragnize output format better
+            idx = 0
+            for product in search_results:
+                print(
+                    _(
+                        f'Product #{idx} - ID:{product.properties["id"]},provider:{product.provider}'
+                    )
+                )
+                idx += 1
+        else:
+            download_products(search_results)
+
+            # TODO: Consider adding a quicklook flag
+            """
+            import matplotlib.pyplot as plt
+            import matplotlib.image as mpimg
+
+            fig = plt.figure(figsize=(10, 8))
+            for i, product in enumerate(search_results, start=1):
+                if i > 7 * 8:
+                    break
+                # This line takes care of downloading the quicklook
+                quicklook_path = product.get_quicklook()
+
+                # Plot the quicklook
+                img = mpimg.imread(quicklook_path)
+                ax = fig.add_subplot(7, 8, i)
+                ax.set_title(i - 1)
+                plt.imshow(img)
+            plt.tight_layout()
+            plt.show()
+            """
+
+
+if __name__ == "__main__":
+
+    try:
+        from eodag import EODataAccessGateway
+        from eodag import setup_logging
+        from eodag.api.product.metadata_mapping import DEFAULT_METADATA_MAPPING
+        from eodag.utils import get_geometry_from_various
+
+        # for debugging -> 3
+        setup_logging(verbose=2)
```

Comment:
You can use GRASS debug messages (`gs.debug()`)

Reply:
I think you have to use gs.debug()

Reply:
Noted, I think I need setup_logging as well to get logs of eodag, e.g. when it is searching? Not sure if there is a way to pipeline the eodag logs into gs.debug...

`setup_logging()` comes from the eodag library to view its logs, that is why I am using it for now.

Reply:
I see. You can match the logging level to the GRASS verbosity / debug level (https://grass.osgeo.org/grass83/manuals/g.message.html), then.
Maybe possible to use [ProgressCallback](https://eodag.readthedocs.io/en/stable/api_reference/utils.html#callbacks)? It should be available in download functions?

Reply:
> You can match the logging level to the GRASS verbosity / debug level (https://grass.osgeo.org/grass83/manuals/g.message.html), then.

I have just added that in this commit: 610f9c9